### PR TITLE
Counter tack part2 - new commands

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/11952

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/integration-CounterTack.yml [hunk 1]
```diff
@@ -974,35 +1038,170 @@ script:
...
+                    'Events Impact': event.get('impact'),
+                    'Events EndpointID': event.get('endpoint_id'),
+                    'Event Type': event.get('event_type'),
+                    'Collected time': event.get('time_stamp'),
+                    'Source process PID': event.get('source_process_pid'),
+                    'Source process name': event.get('source_process_name')
+                })
+
+            context = {
+                'CounterTack.Event': createContext(events, id='ID', keyTransform=underscoreToCamelCase)
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: add `removeNull=True`

File: Integrations/integration-CounterTack.yml [hunk 2]
```diff
@@ -974,35 +1038,170 @@ script:
...
+                'Contents': events,
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': tableToMarkdown('Results of the events search',
+                data, headers, removeNull=True),
+                'EntryContext': context
+            }
+            demisto.results(entry)
+        else:
+            demisto.results("No results found")
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `"` -> `'`

File: Integrations/integration-CounterTack.yml [hunk 3]
```diff
@@ -974,35 +1038,170 @@ script:
...
-            if last_found > current_run:
-                current_run = last_found
+        max_timestamp = last_update_time
+        params = 'behaviors.time_stamp>' + last_update_time
+        behaviors = search_behaviors_request(params)
+
+        for behavior in behaviors.get('results'):
+            # 0 corresponds to never triggered
+
+            time_stamp = behavior.get('behaviors.time_stamp')[:-5]
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please add comment here to describe what you're doing

File: Integrations/integration-CounterTack.yml [hunk 4]
```diff
@@ -1765,14 +1966,203 @@ script:
         - name: file_id
           required: true
           description: The ID of the extracted file
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please add to the description how this can be retrieved

File: Integrations/integration-CounterTack.yml [hunk 5]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+    - name: countertack-search-events
       arguments:
-        - name: file_id
+        - name: expression
           required: true
-          description: The ID of the extracted file
-      description: 'Delete an extracted file from the endpoint.'
+          description: The CQL expression to be used for the search (e.g events.event_type=basic)
+      outputs:
+        - contextPath: CounterTack.Event.Results.SourceProcessTimeStarted
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: not sure why the `Results` level is needed here
please remove it, so for example this will be `CounterTack.Event.SourceProcessTimeStarted`

File: Integrations/integration-CounterTack.yml [hunk 6]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+        - contextPath: CounterTack.Event.Results.SourceProcessPid
+          description: PID of the source process.
+          type: number
+        - contextPath: CounterTack.Event.Results.SourceThreadStartAddress
+          description: Start address of the thread.
+          type: number
+        - contextPath: CounterTack.Event.Results.SourceProcessSid
+          description: User SIDs associated with the process.
+          type: number
+        - contextPath: CounterTack.Event.Results.Id
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: in the code you're counting on it to be `ID`

File: Integrations/integration-CounterTack.yml [hunk 7]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+          type: string
+        - contextPath: CounterTack.Event.Results.ConditionIds
+          description: IDs of the condition triggered by the event.
+        - contextPath: CounterTack.Event.Results.SourceProcessName
+          description: Name of the process that was the source of the event.
+          type: string
+        - contextPath: CounterTack.Event.Results.SourceProcessUser
+          description: User associated with the process
+          type: string
+      description: 'Search for events with CQL expression '
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove unnecessary whitespace at the end

File: Integrations/integration-CounterTack.yml [hunk 8]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+        - name: endpoint_id
+          required: true
+          description: Endpoint ID. Get the ID from the "get-endpoints" command
+        - name: process_id
+          description: 'Process PID. Get the process ID from the "search-events" command'
+        - name: process_name
+          description: The name of the process. Get the process name from the "search-events"
+            command
+      outputs:
+        - contextPath: CounterTack.Endpoint.EndpointIds
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `EndpointIds` -> `ID`

File: Integrations/integration-CounterTack.yml [hunk 9]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+          description: The type of resource or collection this command is being sent to.
+          type: string
+        - contextPath: CounterTack.Endpoint.CommandArg.name
+          description: Name of the process that was terminated.
+          type: string
+        - contextPath: CounterTack.Endpoint.CommandArg.pid
+          description: PID of the process that was terminated.
+          type: number
+        - contextPath: CounterTack.Endpoint.CommandArg
+          description: 'Command Arguments '
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove unnecessary whitespace at the end

File: Integrations/integration-CounterTack.yml [hunk 10]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+        - contextPath: CounterTack.Endpoint.Status
+          description: The status of the command, from (initial, pending, complete, error).
+          type: string
+        - contextPath: CounterTack.Endpoint.CommandName
+          description: The name of the command that is sent.
+          type: string
+        - contextPath: CounterTack.Endpoint.Username
+          description: The username of the user that requested the command.
+          type: string
+        - contextPath: CounterTack.Endpoint.Id
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: if that's a command id, it should be `CounterTack.Endpoint.CommandID`

File: Integrations/integration-CounterTack.yml [hunk 11]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+      description: Terminates all instances of the process identified in the command.
+        Processes can be identified by the PID or process name.
+    - name: countertack-search-hashes
+      arguments:
+        - name: expression
+          required: true
+          description: The CQL expression to be used for the search (e.g hashes.type =
+            md5)
+      outputs:
+        - contextPath: CounterTack.Hash.Results.AvCoverage
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: remove `Results` level

File: Integrations/integration-CounterTack.yml [hunk 12]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+          description: Percentage of AV engines that determined that the Hash is malicious.
+          type: number
+        - contextPath: CounterTack.Hash.Results.Id
+          description: Hashes ID.
+          type: string
+        - contextPath: CounterTack.Hash.Results.Impact
+          description: Impact score for the event in the Hash (1- 100).
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: `1- 100` -> `1-100`

File: Integrations/integration-CounterTack.yml [hunk 13]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+          type: date
+      description: Terminates all instances of the process identified in the command.
+        Processes can be identified by the PID or process name.
+    - name: countertack-search-hashes
+      arguments:
+        - name: expression
+          required: true
+          description: The CQL expression to be used for the search (e.g hashes.type =
+            md5)
+      outputs:
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: need to add `File` context standard

File: Integrations/integration-CounterTack.yml [hunk 14]
```diff
@@ -1765,14 +1966,203 @@ script:
...
+        - contextPath: CounterTack.Hash.Results.Impact
+          description: Impact score for the event in the Hash (1- 100).
+          type: number
+        - contextPath: CounterTack.Hash.Results.Type
+          description: Type of Hash (sha256, md5, or ssdeep).
+          type: string
+        - contextPath: CounterTack.Hash.Results.VtReportLocation
+          description: Report location for Virus Total report.
+          type: string
+      description: 'hashes search using CQL expression '
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please improve description

File: Integrations/integration-CounterTack.yml [hunk 15]
```diff
@@ -1765,14 +1966,203 @@ script:
...
   isfetch: true
   runonce: false
+releaseNotes: "Added new commands to the integration"
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please elaborate which commands were added in the release notes

File: Integrations/integration-CounterTack.yml [hunk 16]
```diff
@@ -85,8 +95,10 @@ script:
             auth=(USERNAME, PASSWORD),
             headers=headers,
             verify=USE_SSL,
-            data=body
+            data=body,
+            params=params
```
Comment (Itay4) [inline Integrations/integration-CounterTack.yml]: please remove `params` if never used

Comment (Shellyber) [reply Integrations/integration-CounterTack.yml]: params is used in fetch incidents command (get behaviors)
